components/style/atomic_refcell.rs

```diff
@@ -39,3 +67,21 @@ impl<T: Default> Default for AtomicRefCell<T> {
         Self::new(T::default())
     }
 }
+
+pub type MappedAtomicRef<'a, T: 'a, U: 'a> = OwningRef<AtomicRef<'a, T>, U>;
+impl<'a, T> AtomicRef<'a, T> {
+    pub fn map<U: ?Sized, F>(orig: Self, f: F) -> MappedAtomicRef<'a, T, U>
```

Comment:
I think I'd expect `Ref::map(Ref::map(..))` to work? Basically, I expect `map` to be an operation from `A<T>` to `A<U>`. I see why it can't be done this way in this case, but I think it's worth a comment?